controllers/nonadmin_controller.go

```diff
@@ -0,0 +1,194 @@
+			corev1.EventTypeNormal,
+			"DeletedNonAdminDeploymentDeployment",
+			fmt.Sprintf("Non admin controller deployment %s/%s deleted", nonAdminDeployment.Namespace, nonAdminDeployment.Name),
+		)
+		return true, nil
+	}
+
+	operation, err := controllerutil.CreateOrUpdate(
+		r.Context,
+		r.Client,
+		nonAdminDeployment,
+		func() error {
+			// Setting Deployment selector if a new object is created, as it is immutable
+			if nonAdminDeployment.ObjectMeta.CreationTimestamp.IsZero() {
+				nonAdminDeployment.Spec.Selector = &metav1.LabelSelector{
+					MatchLabels: nonAdminControlPlaneLabel,
+				}
+			}
+
+			nonAdminImage := r.getNonAdminImage(&dpa)
+			// TODO remove, just for tests
+			log.Info(fmt.Sprintf("NON ADMIN IMAGE: %v", nonAdminImage))
+			r.buildNonAdminDeployment(nonAdminDeployment, nonAdminImage)
+
+			// Setting controller owner reference on the non admin controller deployment
+			return controllerutil.SetControllerReference(&dpa, nonAdminDeployment, r.Scheme)
+		},
+	)
+
+	if err != nil {
+		// TODO needed?
+		if k8serror.IsInvalid(err) {
+			cause, isStatusCause := k8serror.StatusCause(err, metav1.CauseTypeFieldValueInvalid)
+			if isStatusCause && cause.Field == "spec.selector" {
+				log.Info("Found immutable selector from previous deployment, recreating non admin controller deployment")
+				err := r.Delete(r.Context, nonAdminDeployment)
+				if err != nil {
+					return false, err
+				}
+				return r.ReconcileNonAdminController(log)
+			}
+		}
+
+		return false, err
+	}
+
+	if operation != controllerutil.OperationResultNone {
+		r.EventRecorder.Event(
+			nonAdminDeployment,
+			corev1.EventTypeNormal,
+			"NonAdminDeploymentReconciled",
+			fmt.Sprintf("Non admin controller deployment %s/%s was %s", nonAdminDeployment.Namespace, nonAdminDeployment.Name, operation),
+		)
+	}
+	return true, nil
+}
+
+func (r *DPAReconciler) buildNonAdminDeployment(deploymentObject *appsv1.Deployment, image string) {
+	deploymentObject.ObjectMeta.Labels = nonAdminDeploymentLabels
+
+	deploymentObject.Spec = appsv1.DeploymentSpec{
+		Replicas: pointer.Int32(1),
+		Selector: deploymentObject.Spec.Selector,
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: nonAdminControlPlaneLabel,
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Image:           image,
+						ImagePullPolicy: corev1.PullAlways,
+						Name:            nonAdminControllerContainer,
+					},
+				},
+				RestartPolicy:      corev1.RestartPolicyAlways,
+				ServiceAccountName: nonAdminControllerControllerManager,
+			},
+		},
+	}
+}
+
+func (r *DPAReconciler) checkNonAdminEnabled(dpa *oadpv1alpha1.DataProtectionApplication) bool {
+	if dpa.Spec.Features != nil && dpa.Spec.Features.EnableNonAdmin != nil {
+		return *dpa.Spec.Features.EnableNonAdmin
+	}
+
+	return false
+}
+
+func (r *DPAReconciler) getNonAdminImage(dpa *oadpv1alpha1.DataProtectionApplication) string {
+	// TODO is this needed?
```

Comment:
going to remove this TODO, but I think we can think about it in the future

is not as easy as DPA field to change env var in pod?